infra/modules/gcp-tf-runner/main.tf

```diff
@@ -5,29 +5,68 @@
 # times throughout the code base, and includes some hard-coded convention stuff that imho is better
 # to have in one place.
 
+
+# in cloud shell, this seems to return {"email":"", "id":""}
+# in any env, this is NEVER the gcp service account configured via provider block
+# (eg, google.impersonate_service_account = "terraform@...")
```

Comment:
can't we use env variables and make the user explicitly set it in cloud shell as prereq??

export TF_VAR_email=whatever@iam.gserviceaccount.com

Reply:
does that work below the top-level? I would think it would not.

Our problem is that in individual modules, they need to know the identity TF is running as, to grant `ServiceAccountUser` to TF on the SA that TF itself created.  It's a little strange to require this to be passed into the module, as well as tedious because those modules are actually called 2-3 levels deep.